src/Aquifer.API/Endpoints/Resources/Content/NeedsTranslation/List/Endpoint.cs

```diff
@@ -0,0 +1,139 @@
+using Aquifer.API.Common;
+using Aquifer.API.Services;
+using Aquifer.Common.Utilities;
+using Aquifer.Data;
+using Aquifer.Data.Entities;
+using FastEndpoints;
+using Microsoft.Data.SqlClient;
+using Microsoft.EntityFrameworkCore;
+
+namespace Aquifer.API.Endpoints.Resources.Content.NeedsTranslation.List;
+
+public class Endpoint(AquiferDbContext dbContext, IUserService userService) : Endpoint<Request, Response>
+{
+    public override void Configure()
+    {
+        Get("/resources/content/needs-translation");
+        Permissions(PermissionName.ReadResources);
+    }
+
+    public override async Task HandleAsync(Request req, CancellationToken ct)
+    {
+        var user = await userService.GetUserFromJwtAsync(ct);
+        var (hasSearchQuery, isExactSearch) = GetSearchParams(req.SearchQuery);
+        var likeKey = isExactSearch ? "" : "%";
+        var searchParameter = new SqlParameter("SearchQuery", hasSearchQuery ? $"{likeKey}{req.SearchQuery!.Trim('"')}{likeKey}" : "");
+
+        var total = (await dbContext.Database.SqlQueryRaw<int>(BuildQuery(req, user, true, hasSearchQuery, isExactSearch), searchParameter)
+            .ToListAsync(ct)).Single();
+        var resourceContent = total == 0
+            ? []
+            : await dbContext.Database
+                .SqlQueryRaw<ResourceContentResponse>(BuildQuery(req, user, false, hasSearchQuery, isExactSearch), searchParameter)
+                .ToListAsync(ct);
+
+        var response = new Response
+        {
+            ResourceContents = resourceContent,
+            Total = total
```

Comment:
It looks like the SQL query to fetch the total count is only used to populate this property.  Will total be shown in the UI to the user according to the current spec?  And even if so, is it necessary to show a total count in the UI or can we get rid of it to avoid making the additional SQL call altogether?

I'm not opposed to fetching the total if there is a good case to show it to the user.  I just want to make sure it's necessary.

Reply:
👍  Yeah it's being used to determine number of pages available on the frontend